property-scalaz/src/main/scala/testz/property.scala

```diff
@@ -45,34 +44,36 @@ object property {
     [F[_], I]
     (seed: (Array[Int], Int))
     (gen: Int => I)
-    (testGenerator: I => F[List[TestError]])
+    (testGenerator: I => F[TestResult])
     (numTestCases: Int)
-    (implicit F: Applicative[F]): F[List[TestError]] = {
+    (implicit F: Applicative[F]): F[TestResult] = {
     val rng = Well44497a.fromSeed(seed)
-    var acc: F[List[TestError]] = F.pure(Nil)
+    var acc: F[TestResult] = F.pure(Success)
     var i = 0
     while (i < numTestCases) {
-      acc = F.apply2(acc, testGenerator(gen(rng.nextInt)))(_ ++ _)
+      acc = F.apply2(acc, testGenerator(gen(rng.nextInt)))(TestResult.combine)
       i = i + 1
     }
-    acc
+    if (numTestCases == 0) F.pure(Failure.string("No test cases passed to `fromSeed`"))
```

Comment:
`if (i == 0)`?

Reply:
Either does the same thing.

Reply:
Woops, yes. Why is this check necessary? Why not `<= 0`?

Reply:
Guess I could do `<= 0`.